infra/azure/main.tf

```diff
@@ -60,18 +62,50 @@ resource "azurerm_kubernetes_cluster" "vdc" {
 }
 
 resource "azurerm_kubernetes_cluster_node_pool" "vdc_pool" {
-  name = "pool"
+  name                  = "pool"
   kubernetes_cluster_id = azurerm_kubernetes_cluster.vdc.id
-  vm_size = "Standard_D2_v2"
-  vnet_subnet_id = azurerm_subnet.k8s_subnet.id
+  vm_size               = "Standard_D2_v2"
+  vnet_subnet_id        = azurerm_subnet.k8s_subnet.id
 
   enable_auto_scaling = true
 
-  node_count = 1
```

Comment:
This just shouldn't have been here in the first place, since there's already a min and max count set. I think this is what you faced when it said there were changes to be made because the node pool had naturally scaled down to 0.